scripts/zones/Mhaura/npcs/Fyi_Chalmwoh.lua

```diff
@@ -3,19 +3,65 @@
 --  NPC: Fyi_Chalmwoh
 -- !pos -39.273 -16.000 70.126 249
 -----------------------------------
+require("scripts/globals/items")
+require("scripts/globals/quests")
+-----------------------------------
 local entity = {}
 
 entity.onTrade = function(player, npc, trade)
+    local rainingMannequins = player:getQuestStatus(xi.quest.log_id.OTHER_AREAS, xi.quest.id.otherAreas.ITS_RAINING_MANNEQUINS)
+    if rainingMannequins == QUEST_COMPLETED then
+        if
+            (npcUtil.tradeHasExactly(trade, { xi.items.HUME_M_MANNEQUIN }) or
+            npcUtil.tradeHasExactly(trade, { xi.items.HUME_F_MANNEQUIN }) or
+            npcUtil.tradeHasExactly(trade, { xi.items.ELVAAN_M_MANNEQUIN }) or
+            npcUtil.tradeHasExactly(trade, { xi.items.ELVAAN_F_MANNEQUIN }) or
+            npcUtil.tradeHasExactly(trade, { xi.items.TARUTARU_M_MANNEQUIN }) or
+            npcUtil.tradeHasExactly(trade, { xi.items.TARUTARU_F_MANNEQUIN }) or
+            npcUtil.tradeHasExactly(trade, { xi.items.MITHRA_MANNEQUIN }) or
+            npcUtil.tradeHasExactly(trade, { xi.items.GALKA_MANNEQUIN })) and
+            player:getGil() >= 2000
+        then
+            player:startEvent(319, 2, 0, 100000, 2000, 0, 0, 4095, 0)
+        end
+    end
 end
 
 entity.onTrigger = function(player, npc)
-    player:additem({ id = 256, exdata = { [18] = 1 } })
+    local rainingMannequins = player:getQuestStatus(xi.quest.log_id.OTHER_AREAS, xi.quest.id.otherAreas.ITS_RAINING_MANNEQUINS)
+
+    -- Mannequin shop + exchange
+    -- TODO: When you try to buy a new one, mannequins you already own aren't available
+    -- TODO: Check money
+    if rainingMannequins == QUEST_COMPLETED then
+        local alreadyOwnedMask = 0
+        player:startEvent(318, 0, alreadyOwnedMask, 100000, 2000, 0, 0, 0, 0)
+    end
 end
 
 entity.onEventUpdate = function(player, csid, option)
 end
 
 entity.onEventFinish = function(player, csid, option)
+    if csid == 318 or csid == 319 then
```

Comment:
Was just thinking about this block.  Since the items are linear, this could be simplified without even the need for an additional table (We would probably want to sanity check that option is in 1..8, but after that it could be reduced to one line:
```
player:addItem({ id = option + 255 }, exdata = { [18] = option } })
```

Reply:
> Was just thinking about this block. Since the items are linear, this could be simplified without even the need for an additional table (We would probably want to sanity check that option is in 1..8, but after that it could be reduced to one line:
> 
> ```
> player:addItem({ id = option + 255, exdata = { [18] = option } })
> ```

Nope, uncle Teo lesson time: NEVER trust the optioncode in either event finish or event update to actually be legit. they can be faked, so you have to re-validate the items against said code every time.

Reply:
I dunno why my brain decided to be concerned there Clay, the existing code still woudl have that problem AND who the F is cheating at mannequin races anyway. I'm gonna blame my lack of sleep.

Reply:
If that's the case, both versions might have to be rethought.  Do we have a way of knowing which option was selected in onTrigger though?

Reply:
damn son yer too quick, was about to hide my posts

Reply:
> Do we have a way of knowing which option was selected in onTrigger though?

historically we've been using get/setLocalVar to record and read back what the original input was before shenanigans happen.